perfkitbenchmarker/pkb.py

```diff
@@ -417,6 +422,102 @@ def _PrintHelp(matches=None):
         print(FLAGS.module_help(module_name))
 
 
+def _PrintHelpMD(matches=None):
+  """Prints markdown help for flags defined in matching modules. Works just like
+  --helpmatch.
+
+  Eg:
+
+* all flags:
+`./pkb.py --helpmatchmd .*`
+
+* linux benchmarks:
+`./pkb.py --helpmatchmd linux_benchmarks.*`
+
+* specific modules
+`./pkb.py --helpmatchmd iperf`
+
+* windows packages
+`./pkb.py --helpmatchmd windows_packages.*`
+
+* GCP provider:
+`./pkb.py --helpmatchmd providers.gcp.* >> testsuite_docs/providers_gcp.md`
+
+  Args:
+    matches: regex string or None. Filters help to only those whose name
+      matched the regex. If None then all flags are printed.
+  """
+
+  """ example docstring:
+  absl.app:
+  --[no]only_check_args: Set to true to validate args and exit.
+    (default: 'false')
+  --[no]pdb_post_mortem: Set to true to handle uncaught exceptions with PDB post
+    mortem.
+    (default: 'false')
+  --profile_file: Dump profile information to a file (for python -m pstats).
+    Implies --run_with_profiling.
+  --[no]run_with_pdb: Set to true for PDB debug mode
+    (default: 'false')
+  --[no]run_with_profiling: Set to true for profiling the script. Execution will
+    be slower, and the output format might change over time.
+    (default: 'false')
+  --[no]use_cprofile_for_profiling: Use cProfile instead of the profile module
+    for profiling. This has no effect unless --run_with_profiling is set.
+    (default: 'true')
+"""
+
+  # normal helpmatch search from above
+  if not matches:
+    print(FLAGS)
```

Comment:
It seems like if you don't specify a match, you would not generate the markdown and produce the same results as with --helpmatch. That said, it looks like matches is never None (based on the call sites), so this can probably be dropped.

Reply:
Fixed (removed). This check is in _PrintHelp too and seems to suggest that calling `--helpmatch` with no arguments would print(FLAGS) which was the expected behaviour here as well (without markdown). Running `./pkb.py --helpmatch` doesn't produce anything helpful though... should these flags have a default value of `pkb` or `.*` instead of the empty string?

Reply:
I think you could do this, but I think we skip it on the empty string. So that's probably fine.